test/test_datasets_utils.py

```diff
@@ -161,21 +115,21 @@ def create_compressed(root, content="this is the content"):
 
             utils.extract_archive(compressed, temp_dir, remove_finished=True)
 
-            self.assertFalse(os.path.exists(compressed))
+            assert not os.path.exists(compressed)
 
-    def test_extract_archive_defer_to_decompress(self):
+    @pytest.mark.parametrize('extension', [".gz", ".xz"])
+    @pytest.mark.parametrize('remove_finished', [True, False])
+    def test_extract_archive_defer_to_decompress(self, extension, remove_finished, mocker):
         filename = "foo"
-        for ext, remove_finished in itertools.product((".gz", ".xz"), (True, False)):
-            with self.subTest(ext=ext, remove_finished=remove_finished):
-                with unittest.mock.patch("torchvision.datasets.utils._decompress") as mock:
-                    file = f"{filename}{ext}"
-                    utils.extract_archive(file, remove_finished=remove_finished)
-
-                mock.assert_called_once()
-                self.assertEqual(
-                    call_args_to_kwargs_only(mock.call_args, utils._decompress),
-                    dict(from_path=file, to_path=filename, remove_finished=remove_finished),
-                )
+        original_decompress = utils._decompress
+        mocked = mocker.patch("torchvision.datasets.utils._decompress")
+        file = f"{filename}{extension}"
+        utils.extract_archive(file, remove_finished=remove_finished)
+
+        mocked.assert_called_once()
+
+        assert (call_args_to_kwargs_only(mocked.call_args, original_decompress) ==
+                dict(from_path=file, to_path=filename, remove_finished=remove_finished))
```

Comment:
I think we can change this into

```py
        mocked.assert_called_once_with(file, filename, remove_finished=remove_finished)
```

and remove `original_decompress` and also the entire `call_args_to_kwargs_only` function which isn't used anywhere else.

Let's wait for @pmeier  confirmation here ;)

Reply:
Yes, that will make it much cleaner. `call_args_to_kwargs_only` is indeed only referenced here.

Reply:
Yes, we can do this. I've only introduced it, because I really dislike the exact matching, if there are multiple ways that are functionally equal. But since the usage hasn't caught on, I'm ok with removing it.

Reply:
Done!